tests/integration/test_issue_1730.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os
+import sys
+
+import pytest
+
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 7) or sys.version_info[:2] >= (3, 10),
```

Comment:
This test does not run in CI, which just covers 2.7 and 3.10 in ITs. It does work though on `3.{7,8,9}` as tested locally and it is slow (~12s for me) since Pants is big; so it's for the better.